src/backend/dx12/src/conv.rs

```diff
@@ -421,27 +421,82 @@ pub fn map_function(fun: Comparison) -> D3D12_COMPARISON_FUNC {
     }
 }
 
-pub fn map_image_resource_state(access: image::Access, layout: image::ImageLayout) -> D3D12_RESOURCE_STATES {
+pub fn map_buffer_resource_state(access: buffer::Access) -> D3D12_RESOURCE_STATES {
+    // Mutable states
+    if access.contains(buffer::SHADER_WRITE) {
```

Comment:
do we check anywhere that only a single mutable flag is specified?

Reply:
no, I would shift this into the validation layer.